pandas/io/excel.py

```diff
@@ -512,6 +541,36 @@ def _parse_excel(self,
 
         _validate_header_arg(header)
 
+        if (usecols is not None) and (usecols_excel is not None):
+            raise TypeError("Cannot specify both `usecols` and `usecols_excel`"
+                            ". Choose one of them.")
+        # Check if some string in usecols may be interpreted as a Excel
+        # range or positional column
+        elif isinstance(usecols, compat.string_types) and all(
+                x in string.ascii_letters for x in re.split(r',|:', usecols)):
+            warnings.warn("The `usecols` named argument used to refer to "
+                          "Excel ranges and columns as strings was "
+                          "renamed to `usecols_excel`.", UserWarning,
+                          stacklevel=3)
+            usecols_excel = usecols
+            usecols = None
+        # Check if usecols_excel is a list of Excel ranges and columns and if
+        # it is usecols_excel is transformed in a string containing
+        # Excel ranges and columns
+        elif isinstance(usecols_excel, list) and all(
+            isinstance(elem, compat.string_types) and
```

Comment:
see my comment about, if you actually want to accept this, then you are duplicating the above check

Reply:
Are you sure? I have to check if `usecols_excel` is a string like `"A:E,W"` and if it's a list like `["A:E", "W"]`. First I check if `usecols_excel` is the second one and if it's I transform it in the first one. After that I check if it's the first one.

The thing is that `usecols_excel` may not be that list above (second case) and if it's not I also have to check if it's a A1 string. What I can do to avoid the duplicated check in the  special case you cited is place the if for the string in the if else where is the check for `usecols_excel` as a list.